services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx

```diff
@@ -207,114 +175,49 @@ export const StateSubmissionForm = (): React.ReactElement => {
-            }
-
-            setFormDataFromLatestRevision(planFormData)
-
-            //set previous submitted files
-            const documentList = makeDocumentList(submissionAndRevisions)
-            //set document dates
-            const documentDates = makeDateTable(submissionAndRevisions)
-            setDocumentDates(documentDates)
-            if (documentList instanceof Error) {
-                recordJSException(
-                    `StateSubmissionForm: MALFORMATTED_DATA. document list malformatted. ID:
-                    ${submissionAndRevisions.id} Error message: ${documentList.message}`
-                )
-                setFormDataError('MALFORMATTED_DATA')
-                return
-            }
-            setPreviousDocuments(documentList.previousDocuments)
-
-            // set unlock info
-            if (submissionAndRevisions.status === 'UNLOCKED') {
-                const unlockInfo = revision.unlockInfo
-
-                if (unlockInfo) {
-                    setUnlockedInfo({
-                        updatedBy: unlockInfo.updatedBy,
-                        updatedAt: unlockInfo.updatedAt,
-                        updatedReason: unlockInfo.updatedReason,
-                    })
-                } else {
-                    recordJSException(
-                        `StateSubmissionForm: submission in summary has no revision with unlocked information. ID:
-                        ${submissionAndRevisions.id}`
-                    )
-                    setShowPageErrorMessage(
-                        'This may be an unlocked submission that is currently being edited. Please reload the page and try again.'
-                    )
-                }
-            }
-        }
-    }, [submissionAndRevisions])
-
-    if (fetchError) {
-        // This is a sign that we are handling the same error handling logic frontend and backend around invalid status
-        let specificContent: React.ReactElement | undefined = undefined
-        fetchError.graphQLErrors.forEach((err) => {
-            if (err?.extensions?.code === 'WRONG_STATUS') {
-                if (
-                    currentRoute !== 'UNKNOWN_ROUTE' &&
-                    STATE_SUBMISSION_FORM_ROUTES.includes(currentRoute)
-                ) {
-                    specificContent = <ErrorInvalidSubmissionStatus />
-                }
-            }
-        })
+    if (fetchResult.status === 'LOADING') {
+        return (
+            <GridContainer>
+                <Loading />
+            </GridContainer>
+        )
+    }
 
-        return specificContent ?? <GenericErrorPage />
+    if (fetchResult.status === 'ERROR') {
+        recordJSException(fetchResult.error)
+        console.error('Error from API fetch', fetchResult.error)
+        return <GenericErrorPage /> // api failure or protobuf decode failure
     }
 
-    if (!fetchLoading && !submissionAndRevisions) {
+    const { data, formDatas, documentDates, documentLists } = fetchResult
```

Comment:
I'm just agreeing with myself here, but I think this kind of thing shows the power of separating the bit of information we want from the query--the types are much less work when we add it on the back end, and the parsing is easier on the front end.